sdk/compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Upgrade.hs

```diff
@@ -198,19 +325,19 @@ throwIfNonEmpty handleError hm =
               ctxHandler =
                   case ctx of
                     Nothing -> id
-                    Just ctx -> withContextF present ctx
+                    Just ctx -> withContextF present' ctx
           in
-          ctxHandler $ diagnosticWithContextF present err
+          ctxHandler $ diagnosticWithContextF present' err
       _ -> pure ()
 
-checkModuleM :: LF.UpgradedPackageId -> Upgrading LF.Module -> TcUpgradeM ()
-checkModuleM upgradedPackageId module_ = do
+checkModuleM :: [LF.TypeConName] -> LF.UpgradedPackageId -> Upgrading LF.Module -> TcUpgradeM ()
+checkModuleM equalDataTypes upgradedPackageId module_ = do
```

Comment:
This parameter is no longer needed

Reply:
On closer inspection I've realized that the equalDataTypes calculation is buggy for single modules. After toying with ways to fix it I've concluded it'll be a big job regardless - I'll fix it in a future PR, since we don't do structural equality checks anywhere at the moment.

Reply:
https://github.com/digital-asset/daml/issues/19859